### PR TITLE
[TE] Support Email Sending from DashBoard using backend configurations

### PR DESCRIPTION
**Pull request by cecilynie**
Remove unnecessary configurations that leads confusion.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java [hunk 1]
```diff
@@ -362,10 +362,6 @@ public Response sendEmailWithText(
       throw new WebApplicationException("Empty : list of recipients" + toAddr);
     }
 
-    if (Strings.isNullOrEmpty(smtpHost)) {
-      smtpHost = thirdeyeConfiguration.getSmtpHost();
-    }
-
     SmtpConfiguration smtpConfiguration = getSmtpConfiguration(smtpHost, smtpPort);
     if (smtpConfiguration == null) {
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: The return of getSmtpConfiguration() cannot be null. You should add a checker in the function, and return null if the smtpConfiguration isn't valid.

Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: Thanks for pointing this out. Changed in next commit.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java [hunk 2]
```diff
@@ -120,13 +120,15 @@ private SmtpConfiguration getSmtpConfiguration(String smtpHost, Integer smtpPort
...
-      smtpConfiguration.setSmtpPort(smtpPort);
+    } else {
+      if (Strings.isNullOrEmpty(smtpHost)) {
+        return null;
+      } else {
+        smtpConfiguration.setSmtpHost(smtpHost);
+      }
+      if (smtpPort != null) {
+        smtpConfiguration.setSmtpPort(smtpPort);
+      }
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: You can assign 25 as default port.

Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: fixed. thanks!

**Pull request merged**